xcm-support/src/lib.rs

```diff
@@ -75,3 +78,82 @@ impl UnknownAsset for () {
 		Err(DispatchError::Other(NO_UNKNOWN_ASSET_IMPL))
 	}
 }
+
+/// Extracts the `AccountId32` from the passed `location` if the network
+/// matches or is `NetworkId::Any`.
+pub struct RelayChainAccountId32Aliases<Network, AccountId>(PhantomData<(Network, AccountId)>);
+impl<Network: Get<NetworkId>, AccountId: From<[u8; 32]> + Into<[u8; 32]> + Clone>
+	xcm_executor::traits::Convert<MultiLocation, AccountId> for RelayChainAccountId32Aliases<Network, AccountId>
+{
+	fn convert(location: MultiLocation) -> Result<AccountId, MultiLocation> {
+		if let MultiLocation {
+			parents: 1,
+			interior: X1(AccountId32 { id, network }),
+		} = location.clone()
+		{
+			if network == NetworkId::Any || network == Network::get() {
+				return Ok(id.into());
+			}
+		};
+		Err(location)
+	}
+
+	fn reverse(who: AccountId) -> Result<MultiLocation, AccountId> {
+		Ok((
+			1,
+			AccountId32 {
+				id: who.into(),
+				network: Network::get(),
+			},
+		)
+			.into())
+	}
+}
+
+/// Allows execution from `origin` if it is `Parent`.
+pub struct AllowRelayedPaidExecutionFromParent<Network>(PhantomData<Network>);
+impl<Network: Get<NetworkId>> ShouldExecute for AllowRelayedPaidExecutionFromParent<Network> {
+	fn should_execute<Call>(
+		origin: &MultiLocation,
+		message: &mut Xcm<Call>,
+		max_weight: Weight,
+		_weight_credit: &mut Weight,
+	) -> Result<(), ()> {
+		ensure!(origin.contains_parents_only(1), ());
```

Comment:
Is it possible to make this a configurable generic that implements `Contains<MultiLocation>` ?
We have a use-case for allowing sibling chains to send `Transact` instructions to us.